### PR TITLE
feat: migrate test setup to operator-sdk 1.8.0

### PR DESCRIPTION
**Pull request by MatousJobanek**
## Migrates toolchain-e2e repo to operator-sdk 1.8.0

* The most significant changes are in the e2e test setup:
    * migrated most of the code from makefile targets to bash scripts 
    * to install the current operator in the cluster, it drops the usage of the internal ConfigMap - instead of it the logic builds bundle & index images and pushes them to quay
    * building & publishing the operator images is moved to GH Action workflow (we cannot do that in openshift-ci)
    * job in openshift-ci then takes the names of these images and let OLM install them to the cluster
 * additional changes in makefile targets:
     * drops the `${QUAY_NAMESPACE}-` prefix - in some cases we could hit the max number of characters in the route
     * creates ConfigMap `member-operator-config` before installing the second member operator to disable webhook creation
* cleans README from obsolete parts and updates new things
* updates dependencies to use k8s v0.20.2 - as a result:
    * migrates most of the code from `runtime.Object` to `client.Object`, but here is still for an improvement - will be done in next PR
* updated go to 1.16
    * in go.mod
    * in GH Actions workflows  
* renames the default labels, pod names, services, etc... to match the new ones
* the metrics service is using https and is protected by kube-rbac-proxy so the route and related calls had to be modified accordingly 
* as for Dockerfile.tools
    * updated go to 1.16
    * updated oc to the latest 4.7
    * dropped operator-sdk as is not needed in openshift-ci anymore

plus:
* To make at least this e2e test job running, I enabled pairing with more than one repository, so it pairs with all `operator-sdk-1.8.0` PRs

### TODO in next PRs
* disable pairing with multiple PRs
* move the scripts to toolchain-cicd repo
* improve the UX of running the scripts
* clean openshift-ci related config and files
* update the perf test setup 
* update the multi-cluster setup

### CODE REVIEW

**Review by alexeykazakov — APPROVED** [review]

File: dev_install.adoc [hunk 1]
```diff
@@ -16,10 +16,14 @@ Install the link:required_tools.adoc[required tools].
...
-. Make sure that the visibility of the `host-operator`, `member-operator`, `member-operator-webhook` and `registration-service` repositories on quay.io are all set to `public`:
+. Make sure that the visibility of these repositories on quay.io are all set to `public`:
  * https://quay.io/repository/<your-username>/host-operator?tab=settings
+ * https://quay.io/repository/<your-username>/host-operator-bundle?tab=settings
+ * https://quay.io/repository/<your-username>/host-operator-index?tab=settings
  * https://quay.io/repository/<your-username>/member-operator?tab=settings
  * https://quay.io/repository/<your-username>/member-operator-webhook?tab=settings
+ * https://quay.io/repository/<your-username>/member-operator-bundle?tab=settings
+ * https://quay.io/repository/<your-username>/member-operator-index?tab=settings
  * https://quay.io/repository/<your-username>/registration-service?tab=settings
```
Comment (xcoulon) [inline dev_install.adoc:27]:
```suggestion
. Make sure that the visibility of these repositories on quay.io are all set to `public`:
 * https://quay.io/repository/<quay-username>/host-operator?tab=settings
 * https://quay.io/repository/<quay-username>/host-operator-bundle?tab=settings
 * https://quay.io/repository/<quay-username>/host-operator-index?tab=settings
 * https://quay.io/repository/<quay-username>/member-operator?tab=settings
 * https://quay.io/repository/<quay-username>/member-operator-webhook?tab=settings
 * https://quay.io/repository/<quay-username>/member-operator-bundle?tab=settings
 * https://quay.io/repository/<quay-username>/member-operator-index?tab=settings
 * https://quay.io/repository/<quay-username>/registration-service?tab=settings
```

Comment (xcoulon) [reply dev_install.adoc:27]: (to match with `export QUAY_NAMESPACE=<quay-username>` on line 16)

**Pull request merged**